src/main/scala/cromwell/webservice/PerRequest.scala

```diff
@@ -103,6 +103,10 @@ object PerRequest {
 trait PerRequestCreator {
   implicit def actorRefFactory: ActorRefFactory
 
-  def perRequest(r: RequestContext, props: Props, message: AnyRef, timeout: Duration = 1 minutes) =
-    actorRefFactory.actorOf(Props(new WithProps(r, props, message, timeout)))
+  def perRequest(r: RequestContext,
+                 props: Props, message: AnyRef,
+                 timeout: Duration = 1 minutes,
+                 name: String = "Endpoint-" + java.lang.Thread.currentThread.getStackTrace()(1).getMethodName + System.nanoTime()) = {
```

Comment:
I pulled it out a little bit but kept the structure. Since Agora's already doing this (almost, I added the "Endpoint-") IMO we should do the same unless we have a good alternative. And if I had one I'd have used it ;)